packages/webapp/src/delegates/hooks.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { useQuery, useQueryClient } from "react-query";
+
+import {
+    queryHeadDelegate,
+    queryMemberByAccountName,
+    useCurrentMember,
+} from "_app";
+import { EdenMember } from "members";
+import { memberHasRepresentative } from "./api";
+
+export const useHeadDelegate = () => useQuery(queryHeadDelegate);
+
+export const useMyDelegation = () => {
+    const [delegates, setDelegates] = useState<EdenMember[]>([]);
+    const queryClient = useQueryClient();
+    const { data: loggedInMember } = useCurrentMember();
+    const { data: leadRepresentative } = useHeadDelegate();
+
+    useEffect(() => {
+        if (!loggedInMember || !leadRepresentative) return;
+        let nextDelegate = loggedInMember; // should we mutate loggedInMember?
+        const myDelegates: EdenMember[] = [];
+        (async () => {
+            while (
+                nextDelegate.account !== leadRepresentative &&
+                memberHasRepresentative(nextDelegate)
+            ) {
+                myDelegates.push(nextDelegate);
+                const { queryKey, queryFn } = queryMemberByAccountName(
+                    nextDelegate!.representative
+                );
+                nextDelegate = await queryClient.fetchQuery(queryKey, queryFn); // TODO: try...catch
```

Comment:
We can't use hooks (`useQuery`) in loops, and I don't think `useQueries` will work well with this kind of `while` loop. So instead, I query imperatively. (This should be wrapped in a `try...catch`.) Because it's async, and we can't use async...await directly in a hook, we wrap it in an IIFE and use it to build up state (`delegates`) which we ultimately return to the calling component or hook.